### ISSUE TITLE
Error gpu 0: </crypt/xmr-stak/xmrstak/backend/nvidia/nvcc_code/cuda_extra.cu>:256>:256

### ISSUE THREAD

**Issue by Sevilla404** [fireice-uk/xmr-stak#2184]
i upgrade to the latest version of xmr-stak and im getting this error:

[2019-01-29 13:24:43] : Mining coin: masari
[2019-01-29 13:24:43] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda10_0'
WARNING: NVIDIA cannot load backend library: libxmrstak_cuda_backend_cuda10_0.so: cannot open shared object file: No such file or directory
[2019-01-29 13:24:43] : NVIDIA: try to load library 'xmrstak_cuda_backend_cuda9_2'
NVIDIA: found 2 potential device's
[2019-01-29 13:24:44] : Starting NVIDIA GPU thread 0, no affinity.
CUDA [9.1/8.0] GPU#0, device architecture 61: "GeForce GTX 1050 Ti"... device init succeeded
[2019-01-29 13:24:44] : Starting NVIDIA GPU thread 1, no affinity.
CUDA [9.1/8.0] GPU#1, device architecture 30: "GeForce GTX 770"... device init succeeded
[2019-01-29 13:24:44] : NVIDIA: use library 'xmrstak_cuda_backend_cuda9_2'
[2019-01-29 13:24:44] : Starting 1x thread, affinity: 0.
[2019-01-29 13:24:44] : hwloc: memory pinned
[2019-01-29 13:24:44] : Starting 1x thread, affinity: 1.
[2019-01-29 13:24:44] : hwloc: memory pinned
[2019-01-29 13:24:44] : Starting 1x thread, affinity: 2.
[2019-01-29 13:24:44] : hwloc: memory pinned
[2019-01-29 13:24:44] : Fast-connecting to 192.168.0.225:3333 pool ...
[2019-01-29 13:24:44] : Pool 192.168.0.225:3333 connected. Logging in...
[2019-01-29 13:24:45] : Difficulty changed. Now: 40500.
[2019-01-29 13:24:45] : Pool logged in.
[CUDA] Error gpu 1: </crypt/xmr-stak/xmrstak/backend/nvidia/nvcc_code/cuda_extra.cu[CUDA] Error gpu 0: </crypt/xmr-stak/xmrstak/backend/nvidia/nvcc_code/cuda_extra.cu>:256>:256

terminate called after throwing an instance of 'std::runtime_errorterminate called recursively

nvidia.txt
"gpu_threads_conf" :
[
  // gpu: GeForce GTX 1050 Ti architecture: 61
  //      memory: 3981/4040 MiB
  //      smx: 6
  { "index" : 0,
    "threads" : 64, "blocks" : 21,
    "bfactor" : 8, "bsleep" :  10,
    "affine_to_cpu" : false, "sync_mode" : 3,
    "mem_mode" : 1,
  },
  // gpu: GeForce GTX 770 architecture: 30
  //      memory: 1946/1998 MiB
  //      smx: 8
  { "index" : 1,
    "threads" : 36, "blocks" : 24,
    "bfactor" : 0, "bsleep" :  0,
    "affine_to_cpu" : false, "sync_mode" : 3,
    "mem_mode" : 1,
  },

i read some post and i belive its about bfactor of cards but i cant find the correct combination.

**Comment by bymaximus** [comment]
Im getting this error also, gtx970, tried cuda 9, 9.1 and 10, all them compiles without error. Benchmark works, it uses the GPU but throws this line 256 error when mining.

**Comment by Spudz76** [comment]
These are my settings for GTX970 / CUDA 10.0 / latest driver
Tuned to stay out of my way a bit (desktop daily driver box)
```
  // gpu: GeForce GTX 970 architecture: 52
  //      memory: 3204/4096 MiB
  //      smx: 13
  //  "threads" : 4, "blocks" : 104,
  //  "bfactor" : 6, "bsleep" :  25,
  //  "affine_to_cpu" : false, "sync_mode" : 3,
  //  "mem_mode" : 1,
  { "index" : 0,
    "threads" : 4, "blocks" : 208,
    "bfactor" : 8, "bsleep" :  100,
    "affine_to_cpu" : 1, "sync_mode" : 3,
    "mem_mode" : 1,
  },
```
Clocking profile:
`"C:\Program Files (x86)\NVIDIA Corporation\NVInspector\nvidiaInspector.exe" -setBaseClockOffset:0,0,44 -setMemoryClockOffset:0,0,440 -setOverVoltage:0,0 -setPowerTarget:0,66 -setTempTarget:0,0,79`
```
HASHRATE REPORT - NVIDIA
| ID |    10s |    60s |    15m |
|  0 |  449.1 |  451.0 |  451.5 |
```
Pretty strong for limited wattage, really.

```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.91       Driver Version: 418.91       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 970    WDDM  | 00000000:01:00.0  On |                  N/A |
| 30%   72C    P0   112W / 112W |   2486MiB /  4096MiB |    100%      Default |
+-------------------------------+----------------------+----------------------+
```

Guess my driver is CUDA 10.1 actually, still compiled with the normal first release Toolkit though...

**Comment by Spudz76** [comment]
I have triggered that error in the past with too much clocking.  Note changing currency changes the work thus the same max clock may not work across all algos.  For example cn-v8 tended to fail hard on users old cn-v7 bleeding-edge settings, especially core being too high.  Memory clock too high usually just throws invalid results versus the whole exec kernel stack exploding.

**Comment by bymaximus** [comment]
No success for me with those settings and default clocking values. I will try other ways after work. Any idea why benchmark works but mining not?

**Comment by bymaximus** [comment]
Oh boy, even on my ubuntu, getting same error. Tried other algos too.

```
[2019-02-20 00:02:37] : NVIDIA: try to load library 'xmrstak_cuda_backend'
NVIDIA: found 1 potential device's
[2019-02-20 00:02:37] : Starting NVIDIA GPU thread 0, no affinity.
CUDA [9.1/9.1] GPU#0, device architecture 52: "GeForce GTX 970"... device init succeeded
[2019-02-20 00:02:37] : NVIDIA: use library 'xmrstak_cuda_backend'
[2019-02-20 00:02:38] : Difficulty changed. Now: 2000.
[2019-02-20 00:02:38] : Pool logged in.
[2019-02-20 00:02:38] : New block detected.
DEBUG DEV 0 LEN 108 INPUT 103023104[CUDA] Error gpu 0: </xmr-stak-latest/xmrstak/backend/nvidia/nvcc_code/cuda_extra.cu>:257
terminate called after throwing an instance of 'std::runtime_error'
  what():  [CUDA] Error: invalid argument
Aborted (core dumped)

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 390.77                 Driver Version: 390.77                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 970     Off  | 00000000:06:00.0  On |                  N/A |
| 58%   39C    P8    17W / 232W |    266MiB /  4041MiB |      1%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1450      G   /usr/lib/xorg/Xorg                           137MiB |
|    0      1649      G   /usr/bin/gnome-shell                         125MiB |
+-----------------------------------------------------------------------------+
```

**Comment by Spudz76** [comment]
That probably hasn't been expanded for the extended-uncle on the job packet, yet.  AMD is (was?) similar.

Previous job length limit was 84 bytes and must be re-expanded everywhere to 108 (probably 124 similar to what one of the other miners did to patch it).  Just in case some dumb coin wants to do five uncles or dumb garbage and changes the cryptonight standard job size again...

The benchmark generates test packets in standard format with no extra uncle / 84 byte format so it works fine (and the algo itself works fine everywhere).  But when the pool provides a job that is 108 bytes nothing lines up and the host-to-gpu memory copy fails due to mismatched sizes. (line 256 is a that memcpy).

Code needs repaired for larger jobsize before it will work.

**Comment by bymaximus** [comment]
Seems thats true, I just went to debug and saw the job data from the pool.
```
RECV: {"id":1,"jsonrpc":"2.0","error":null,"result":{"id":"2b6305e4-aeeb-4571-bffb-ae1bc299686f","job":{"blob":"0808cdaeb3e305e26d6a3d6c5a7e5d49f9f3b7b567f417d27ac0e6c1ed15a3e6b918d675a7f64800000000d1d5d70686eb4f7e655cce8e18aa242c48f1de169c3a7fae22d7e7a7c2e39d21020000000000000000000000000000000000000000000000000000000000000000","job_id":"SRG8i97+4r3DPvAfdF5jhMZmOm6N","target":"711b0d00","id":"2b6305e4-aeeb-4571-bffb-ae1bc299686f"},"status":"OK"}}
```

Its failing for masari, in this case this big job length. Some other coins I have tested are working fine, lost 3 days on this since I currently only my masari, fuc....... :unamused:

**Comment by Spudz76** [comment]
xmrig was patched for it last week or so, try that until it gets patched here...

**Comment by Blitss** [comment]
Same here

**Comment by Spudz76** [comment]
No, it was more around [this](https://github.com/xmrig/xmrig/issues/913)

The pipelines in xmr-stak are sized for the normal CN PoW blob, not one where someone decided to just add bytes to it.  Changing pipeline size means fixing it everywhere the work-pipe goes (sockets, executor, all three backends... and their kernels).

I would just expand it to 256byte so that it never ever ever happens again.  Who cares about moving around (or overlooking, really) the additional expansion zone.  However some of how the sizing works out is based on hitting good memory access alignments, for speed and efficiency (or cache utilization too).

**Comment by psychocrypt** [comment]
fixed with #2296
